src/fermilib/utils/_jellium_trotter_error.py

```diff
@@ -0,0 +1,461 @@
+    return commutator
+
+
+def double_commutator(op1, op2, op3, indices2=None, indices3=None,
+                      is_hopping_operator2=None, is_hopping_operator3=None):
+    """Return the double commutator [op1, [op2, op3]].
+
+    Assumes the operators are from the jellium Hamiltonian.
+
+    Args:
+        op1, op2, op3 (FermionOperators): operators for the commutator.
+        indices2, indices3 (set): The indices op2 and op3 act on.
+        is_hopping_operator2 (bool): Whether op2 is a hopping operator.
+                                     Similarly for is_hopping_operator3.
+    """
+    if is_hopping_operator2 and is_hopping_operator3:
+        indices2 = set(indices2)
+        indices3 = set(indices3)
+        intersection, = indices2.intersection(indices3)
+
+        indices2.remove(intersection)
+        indices3.remove(intersection)
+
+        index2, = indices2
+        index3, = indices3
+        coeff2 = op2.terms[list(op2.terms)[0]]
+        coeff3 = op3.terms[list(op3.terms)[0]]
+        commutator23 = (
+            FermionOperator(((index2, 1), (index3, 0)), coeff2 * coeff3) +
+            FermionOperator(((index3, 1), (index2, 0)), -coeff2 * coeff3))
+    else:
+        commutator23 = commutator(op2, op3)
+
+    return commutator(op1, commutator23)
+
+
+def trivially_double_commutes_more_info(indices_alpha=None, indices_beta=None,
+                                        indices_alpha_prime=None,
+                                        is_hopping_operator_alpha=None,
+                                        is_hopping_operator_beta=None,
+                                        is_hopping_operator_alpha_prime=None):
+    """Determine if [op_a, [op_b, op_a_prime]] is trivially zero.
+
+    Assumes the operators are FermionOperators from the jellium
+    Hamiltonian. They are determined by the indices they act on and by
+    whether they are hopping operators. a, b, and a_prime are shorthands
+    for alpha, beta, and alpha_prime.
+
+    Args:
+        indices_alpha (set): The indices term_alpha acts on.
+                             Defined similarly for beta and alpha_prime.
+        is_hopping_operator_alpha (bool): Whether term_alpha is a
+                                          hopping operator. Similarly
+                                          for beta, alpha_prime.
+    """
+    return not ((is_hopping_operator_beta or
+                 is_hopping_operator_alpha_prime) and
+                (len(indices_beta.intersection(indices_alpha_prime)) == 1) and
+                indices_alpha.intersection(
+                    indices_beta.union(indices_alpha_prime)))
+
+
+def trivially_commutes_jellium(term_a, term_b):
+    """Determine whether the given terms trivially commute.
+
+    Assumes the terms are single-term FermionOperators from the jellium
+    Hamiltonian.
+
+    Args:
+        term_a, term_b (FermionOperator): Single-term FermionOperators.
+    """
+    term_op_a, = term_a.terms.keys()
```

Comment:
"term_op_a" is a terrible name. What does that mean? term operator a? The idea of software engineering is that you are writing the code so that a total stranger can look at this and understand what is going on, as quickly and intuitively as possible.